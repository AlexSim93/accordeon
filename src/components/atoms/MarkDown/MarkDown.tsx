import ReactMarkdown from 'react-markdown';
import styled from 'styled-components';
import { space, color, typography } from 'styled-system';

export const MarkDown = styled(ReactMarkdown)`
  & > p {
    margin: 0;
    ${space}
    ${color}
    ${typography}
  }
`;
