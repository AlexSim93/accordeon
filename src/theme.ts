const breakpoints = ['768px', '1440px'];

export default {
  breakpoints,
  fontSizes: {
    xxs: '10px',
    xs: '12px',
    s: '14px',
    m: '16px',
    ml: '20px',
    l: '24px',
    xl: '32px',
    xxl: '40px',
    xxxl: '56px',
  },
  space: { // s 8, m 16, l 24
    xxxxs: '4px',
    xxxs: '8px',
    xxs: '12px',
    xs: '16px',
    s: '24px',
    sm: '32px',
    m: '40px',
    ml: '48px',
    l: '56px',
    xl: '64px',
    xxl: '72px',
    xxxl: '140px',
    xxxxl: '200px',
  },
  mediaQueries: {
    small: `@media screen and (min-width: ${breakpoints[0]})`,
    medium: `@media screen and (min-width: ${breakpoints[1]})`,
  },
  colors: {
    white: '#FFFFFF', // title text
    blackPearlLighten48: 'rgba(8, 13, 38, 0.48)', // border
    medianGrey: '#A7A7C8', // description text
    darkGreyLighten16: 'rgba(103, 107, 139, 0.16)', // accordion bg
    neutralDarkGrey: '#C4C4C4', // icon
    darkBlue: '#080D26', // page bg
    dodgerBlue: '#564FFF', // button bg
  },
  fonts: {
    Montserrat: '"Montserrat", sans-serif',
  },
  radii: {
    s: '4px',
  },
};
