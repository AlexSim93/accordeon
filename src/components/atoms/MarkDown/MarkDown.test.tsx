import React from "react";
import renderer from "react-test-renderer";
import { MarkDown } from "./MarkDown";
import "jest-styled-components";
import { ThemeProvider } from "styled-components";
import theme from "../../../theme";

describe("MarkDown renders correctly", () => {
  test('MarkDown renders correctly with type=Chevron fill=black and size="m"', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <MarkDown>It's a test task for **overgear**</MarkDown>
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
