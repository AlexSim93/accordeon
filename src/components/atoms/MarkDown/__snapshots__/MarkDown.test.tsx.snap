// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkDown renders correctly MarkDown renders correctly with type=Chevron fill=black and size="m" 1`] = `
.c0 > p {
  margin: 0;
}

<div
  className="c0"
>
  <p>
    It's a test task for 
    <strong>
      overgear
    </strong>
  </p>
</div>
`;
